app/backend/src/clj/teet/integration/vektorio/vektorio_client.clj

```diff
@@ -89,12 +99,18 @@
 
 (defn add-user-to-project!
   "Add the given user to the vektorio project"
-  [vektor-conf project-id user-id]
+  [conn user vektor-conf project-id user-id]
   (assert (some? project-id) "Must specify the project to which the user is added")
   (assert (some? user-id) "Must specify the user which is added to the project")
   (log/info "Adding vektorio user" user-id "to vektorio project:" project-id)
-  (vektor-post! vektor-conf {:endpoint (str "projects/" project-id "/users")
-                             :payload {:userId user-id}}))
+  (when
+    (empty?
+      (d/q '[:find ?pid :where [?u :vektorio/granted-projects ?pid] :in $ ?u ?pid]
+         (d/db conn) (user-model/user-ref user) project-id))
+      (vektor-post! vektor-conf {:endpoint (str "projects/" project-id "/users")
+                             :payload {:userId user-id}})
+             (d/transact conn {:tx-data [{:db/id (:db/id user)
+                                          :vektorio/granted-projects project-id}]})))
```

Comment:
Continuing with the Datomic dependency, if the addition is successful, the calling code can transact the project id to Datomic. This function should only be a wrapper for the HTTP request.